client/apps/shipping-label/redux-middleware.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal dependencies
+ */
+//from calypso
+import {
+	WOOCOMMERCE_SERVICES_SHIPPING_LABEL_EXIT_PRINTING_FLOW,
+} from 'woocommerce/woocommerce-services/state/action-types';
+
+const middlewareActions = {
+	[  WOOCOMMERCE_SERVICES_SHIPPING_LABEL_EXIT_PRINTING_FLOW ]: () => {
```

Comment:
Using the `WOOCOMMERCE_SERVICES_SHIPPING_LABEL_PURCHASE_RESPONSE` action (and checking that there isn't any `error`) would be more efficient, and will finish fetching even before the user closes the modal most of the time.